src/ifp/src/InitFloorplan.cc

```diff
@@ -360,11 +389,53 @@ void InitFloorplan::updateVoltageDomain(dbSite* site,
   }
 }
 
+std::set<dbSite*> InitFloorplan::getSites() const
+{
+  std::set<dbSite*> sites;
+
+  // loop over all instantiated cells in the block
+  for (dbInst* inst : block_->getInsts()) {
+    dbMaster* master = inst->getMaster();
+    // get the site of the master
+    auto site = master->getSite();
+    if (site == nullptr) {
+      continue;
+    }
+    sites.insert(site);
+  }
+
+  std::vector<dbSite*> site_vec{sites.begin(), sites.end()};
+  sort(site_vec.begin(), site_vec.end(), [](dbSite* a, dbSite* b) {
+    return a->getHeight() < b->getHeight();
+  });
+
+  // smallest site height
+  double min_site_height = site_vec[0]->getHeight();
+  double factor = 1.0;
+  for (auto site : site_vec) {
+    double site_height = site->getHeight();
+    factor = site_height / min_site_height;
+    // check if the site height is a multiple of the smallest site height
+    if (std::ceil(factor) != std::floor(factor)) {
+      logger_->error(
+          IFP,
+          228,  // TODO: add error code
+          "site height {} is not a multiple of the smallest site height {}",
+          site_height,
+          min_site_height);
+      // erase from the set if the site height is not a multiple of the smallest
+      sites.erase(site);
```

Comment:
you'll never get here as error() throws an exception.  The erase wouldn't be safe while iterating anyways